### ISSUE TITLE
flutter create --template=package creates platform folders 

### ISSUE THREAD

**Issue by thisisyusub** [flutter/flutter#119844]
## Steps to Reproduce

1. Execute `flutter create --template=package package_name` 

**Expected results:** 
When we create package, I assume that it should only create dart files.

**Actual results:** 
But unfortunately, it creates all platform folders - `android`, `ios`, `macos` and etc. I have tried to create package from Android studio, VS Code and even with flutter command.
<img width="930" alt="Screenshot 2023-02-02 at 22 44 03" src="https://user-images.githubusercontent.com/44983544/216420353-a1f2c448-6df6-48d9-a56e-28459bc68deb.png">

<img width="749" alt="Screenshot 2023-02-02 at 22 44 51" src="https://user-images.githubusercontent.com/44983544/216420520-48d4ed0a-ce2f-4829-b3ff-2d79926bd7cf.png">

<details>
  <summary>Logs</summary>

```
// this is modular app (contains different flutter packages, web and android/ios app)
Analyzing Inside...                                                     
No issues found! (ran in 3.3s)
```

```
[✓] Flutter (Channel stable, 3.7.0, on macOS 13.1 22C65 darwin-arm64 (Rosetta),
    locale en-AZ)
    • Flutter version 3.7.0 on channel stable at
      /Users/kanza/Development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision b06b8b2710 (10 days ago), 2023-01-23 16:55:55 -0800
    • Engine revision b24591ed32
    • Dart version 2.19.0
    • DevTools version 2.20.1

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
    • Android SDK at /Users/kanza/Library/Android/sdk
    • Platform android-33, build-tools 33.0.1
    • Java binary at: /Users/kanza/Library/Application
      Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/221.6008.13.2211.9477386
      /Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      11.0.15+0-b2043.56-8887301)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14C18
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.1)
    • Android Studio at /Users/kanza/Library/Application
      Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/221.6008.13.2211.9477386
      /Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      11.0.15+0-b2043.56-8887301)

[!] Android Studio
    • Android Studio at /Users/kanza/Desktop/Android Studio Preview.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to find bundled Java version.
    • Try updating or re-installing Android Studio.

[✓] IntelliJ IDEA Community Edition (version 2022.3.2)
    • IntelliJ at /Users/kanza/Library/Application
      Support/JetBrains/Toolbox/apps/IDEA-C/ch-0/223.8617.56/IntelliJ IDEA
      CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.74.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.58.0
```

</details>

**Comment by lucamenor** [comment]
Same problem here.

**Comment by guidezpl** [comment]
Related https://github.com/flutter/flutter/issues/116320

**Comment by Terranic** [comment]
They may be needed in case of platform dependent code (e.g. platform channels) -- don´t they? So a full template needs to setup all platform folders.

**Comment by guidezpl** [comment]
> They may be needed in case of platform dependent code (e.g. platform channels) -- don´t they? So a full template needs to setup all platform folders.

That would be true for plugins, which are effectively packages + native code. Packages are generally pure Dart.

**Comment by maRci002** [comment]
Minimal dart package:
`dart create --template=package my_package`

You can add flutter sdk in `pubspec.yaml`
```yaml
dependencies:
  flutter:
    sdk: flutter
```

**Comment by darshankawar** [comment]
Thanks for the report. Seeing same behavior as reported.

[This](https://docs.flutter.dev/development/packages-and-plugins/developing-packages#dart) document gives overview of some of the files generated using the said command, but nothing much on the folders created with it.

With [this](https://docs.flutter.dev/development/packages-and-plugins/developing-packages#step-1-create-the-package-1) though for plugin to create, we specify platforms for which the plugin need to support and hence the respective folders are created.

<details>
<summary> stable, master flutter doctor -v </summary>

```
[!] Flutter (Channel stable, 3.7.0, on macOS 12.2.1 21D62 darwin-x64, locale
    en-GB)
    • Flutter version 3.7.0 on channel stable at
      /Users/dhs/documents/fluttersdk/flutter
    ! Warning: `flutter` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/flutter, which is not inside
      your current Flutter SDK checkout at
      /Users/dhs/documents/fluttersdk/flutter. Consider adding
      /Users/dhs/documents/fluttersdk/flutter/bin to the front of your path.
    ! Warning: `dart` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/dart, which is not inside your
      current Flutter SDK checkout at /Users/dhs/documents/fluttersdk/flutter.
      Consider adding /Users/dhs/documents/fluttersdk/flutter/bin to the front
      of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision b06b8b2710 (28 hours ago), 2023-01-23 16:55:55 -0800
    • Engine revision b24591ed32
    • Dart version 2.19.0
    • DevTools version 2.20.1
    • If those were intentional, you can disregard the above warnings; however
      it is recommended to use "git" directly to perform update checks and
      upgrades.

[!] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! Flutter recommends a minimum Xcode version of 13.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (5 available)
    • SM G975F (mobile)       • RZ8M802WY0X • android-arm64   • Android 11 (API 30)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1 18D61
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 98.0.4758.80

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.

[!] Flutter (Channel master, 3.8.0-3.0.pre.20, on macOS 12.2.1 21D62 darwin-x64,
    locale en-GB)
    • Flutter version 3.8.0-3.0.pre.20 on channel master at
      /Users/dhs/documents/fluttersdk/flutter
    ! Warning: `flutter` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/flutter, which is not inside
      your current Flutter SDK checkout at
      /Users/dhs/documents/fluttersdk/flutter. Consider adding
      /Users/dhs/documents/fluttersdk/flutter/bin to the front of your path.
    ! Warning: `dart` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/dart, which is not inside your
      current Flutter SDK checkout at /Users/dhs/documents/fluttersdk/flutter.
      Consider adding /Users/dhs/documents/fluttersdk/flutter/bin to the front
      of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision bc45b18589 (2 hours ago), 2023-02-02 21:37:03 -0500
    • Engine revision 2696fff871
    • Dart version 3.0.0 (build 3.0.0-196.0.dev)
    • DevTools version 2.21.1
    • If those were intentional, you can disregard the above warnings; however
      it is recommended to use "git" directly to perform update checks and
      upgrades.

[!] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! Flutter recommends a minimum Xcode version of 13.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (5 available)
    • SM G975F (mobile)       • RZ8M802WY0X • android-arm64   • Android 11 (API 30)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1 18D61
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 98.0.4758.80

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.



```

</details>

**Comment by thisisyusub** [comment]
Thanks for comments. 
@darshankawar so I understood it is okay for packages to generate different platform packages, right?

**Comment by thisisyusub** [comment]
> 

Yes this is true but this package can't depend on Flutter related things (for example, creating any feature ui with material and cupertino components in this package and so on), but in the normal case we would create package that can use Flutter related codes too.

**Comment by thisisyusub** [comment]
> 

Yes this is true but this package can't depend on Flutter related things (for example, creating any feature ui with material and cupertino components in this package and so on), but in the normal case we would create package that can use Flutter related codes too.

**Comment by thisisyusub** [comment]
> Minimal dart package: `dart create --template=package my_package`
> 
> You can add flutter sdk in `pubspec.yaml`
> 
> ```yaml
> dependencies:
>   flutter:
>     sdk: flutter
> ```

> 

Yes this is true but this package can't depend on Flutter related things (for example, creating any feature ui with material and cupertino components in this package and so on), but in the normal case we would create package that can use Flutter related codes too.

**Comment by christopherfujino** [comment]
Although I didn't know about this template, it seems a valid use case where you want to author a package that calls dart:ui.

**Comment by eximius313** [comment]
if this platform specific packages are supposed to be generated, please allow at least to configure them via `--platforms`, because not everybody needs every platform

**Comment by Rexios80** [comment]
The `package` template has always been to create reusable dart-only flutter code. Adding platform-specific files would make the project a `plugin` instead. This template only recently started generating platform folders. The files generated seem to be files that would be ignored anyways even if they were supposed to be there. Except they aren't ignored because this template isn't supposed to be generating them.

<img width="353" alt="Screenshot 2023-03-15 at 6 26 57 PM" src="https://user-images.githubusercontent.com/7896519/225457452-af59e064-8308-431c-a79c-3d9ed9fac6dd.png">

**Comment by abdulrojakdev** [comment]
why is this issue not resolved in Flutter version 3.10.0 ?

**Comment by christopherfujino** [comment]
> why is this issue not resolved in Flutter version 3.10.0 ?

because it was merged after Flutter 3.10 branched.
